### DISCUSSION TITLE
Limiting Queue Depth for Blocking Thread Pool

### DISCUSSION THREAD

**Discussion by joshleeb** [tokio-rs/tokio#6347] [Q&A] [answered]
In the docs for [`task::spawn_blocking`](https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html) it says that if all the threads in the blocking thread pool are busy, then the tasks are put in a queue. And I see that the upper limit of the blocking thread pool can be configured for the runtime with [`Builder::max_blocking_threads`](https://docs.rs/tokio/latest/tokio/runtime/struct.Builder.html#method.max_blocking_threads).

Q. _Is there a way to configure, or otherwise limit, the max depth of the queue containing tasks awaiting the blocking thread pool?_

Thanks! 🙏

**Comment by Darksonn** [comment, marked as answer]
No, the queue is unbounded. If you need custom limits on `spawn_blocking` tasks, then you can use a [`Semaphore`](https://docs.rs/tokio/latest/tokio/sync/struct.Semaphore.html). See its examples for ideas. You should be able to use `try_acquire` to fail spawning when there are too many tasks.

**Reply by joshleeb** [reply]
Perfect, thank you! And thank you for the quick response 🎉
